homeassistant/components/point/config_flow.py

```diff
@@ -1,193 +1,51 @@
 """Config flow for Minut Point."""
 
-import asyncio
-from collections import OrderedDict
+from collections.abc import Mapping
 import logging
 from typing import Any
 
-from pypoint import PointSession
-import voluptuous as vol
-
-from homeassistant.components.http import KEY_HASS, HomeAssistantView
-from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
-from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
-from homeassistant.core import callback
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
 
 from .const import DOMAIN
 
-AUTH_CALLBACK_PATH = "/api/minut"
-AUTH_CALLBACK_NAME = "api:minut"
-
 DATA_FLOW_IMPL = "point_flow_implementation"
 
-_LOGGER = logging.getLogger(__name__)
-
-
-@callback
-def register_flow_implementation(hass, domain, client_id, client_secret):
-    """Register a flow implementation.
-
-    domain: Domain of the component responsible for the implementation.
-    name: Name of the component.
-    client_id: Client id.
-    client_secret: Client secret.
-    """
-    if DATA_FLOW_IMPL not in hass.data:
-        hass.data[DATA_FLOW_IMPL] = OrderedDict()
 
-    hass.data[DATA_FLOW_IMPL][domain] = {
-        CONF_CLIENT_ID: client_id,
-        CONF_CLIENT_SECRET: client_secret,
-    }
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle Minut Point OAuth2 authentication."""
 
+    DOMAIN = DOMAIN
 
-class PointFlowHandler(ConfigFlow, domain=DOMAIN):
-    """Handle a config flow."""
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
 
-    VERSION = 1
+    async def async_step_import(self, data: dict[str, Any]) -> ConfigFlowResult:
+        """Handle import from YAML."""
+        return await self.async_step_user()
```

Comment:
You don't have an import flow startpoint?

Reply:
No it doesn't seem necessary. This works fine in my tests.